source/nodejs/adaptivecards-designer/src/containers/theme-supported-host-container.ts

```diff
@@ -0,0 +1,53 @@
+import * as Adaptive from "adaptivecards";
+import { HostContainer } from "./host-container";
+
+export enum ColorTheme {
+    Light = "Light",
+    Dark = "Dark"
+}
+
+export abstract class ThemeSupportedHostContainer extends HostContainer {
+    private _colorTheme: ColorTheme;
+    private _cssFileName: string;
+    private _hostConfigLight: any;
+    private _hostConfigDark: any;
+    private _lightBackground: string;
+    private _darkBackground: string;
+
+    constructor(
+        name: string,
+        cssFileName: string,
+        hostConfigLight: any,
+        hostConfigDark: any,
+        lightBackground: string,
+        darkBackground: string
+    ) {
+        super(name, `containers/${cssFileName}-light.css`);
```

Comment:
I prefer a function because it's easy to introduce bug when the same code logic is copied in multiple places.

```
getCurrentCssFileName()
```
and then
```
 super(name, getCurrentCssFileName())
```
and
```
this.styleSheet = getCurrentCssFileName();
```

Of course, you can do it GetCssFileName(theme: Theme) too

Reply:
I just updated the code to have another method, `getCurrentStyleSheet` that is implemented by `MultiThemeHostContainer`  and `SingleThemeHostContainer`. Let me know what you think!